pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -520,6 +520,16 @@ func (host *BareMetalHost) WasProvisioned() bool {
 // WasExternallyProvisioned returns true when we think something else
 // is managing the image running on the host.
 func (host *BareMetalHost) WasExternallyProvisioned() bool {
+	// We do not want provisioned hosts to appear to be externally
+	// provisioned when we start deprovisioning them.
+	if host.Status.Provisioning.State == StateProvisioned {
+		return false
+	}
+	if host.Status.Provisioning.State == StateDeprovisioning {
+		return false
+	}
+	// Truly externally provisioned hosts have an image and no
+	// consumer.
 	if host.Spec.Image == nil && host.Spec.ConsumerRef != nil {
```

Comment:
I think a better test would be that the Host was externally provisioned if:

* The state is already `StateExternallyProvisioned`; OR
* All of:
  * The state is `StateRegistering`, AND
  * No image is specified, AND
  * There is a consumer

That's actually what our [state diagram](https://github.com/metal3-io/baremetal-operator/blob/master/docs/baremetalhost-states.md) says we do.